### PR TITLE
first pass at a build-addon script; fixes #105

### LINKED ISSUE #105: Hook Firefox Notes up to Test Pilot
1) We need to make this repository match the spec at https://github.com/mozilla/testpilot/blob/master/docs/development/hosting.md

2) @relud to add this repository to the watcher

### PR DESCRIPTION
**Pull request by clouserw**
This isn't ready to land yet, but is close.  We still need to add permissions on AMO (I wasn't sure if this has been signed before or not) as well as address #122

### CODE REVIEW

File: bin/build-addon.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    exit 1
+
+else
+    npm run build
+    rm -f ./web-ext-artifacts/*.xpi
+    ./node_modules/.bin/web-ext sign \
+    --source-dir src \
+    --api-key $TESTPILOT_AMO_USER \
+    --api-secret $TESTPILOT_AMO_SECRET
+    mv ./web-ext-artifacts/*.xpi ./signed-addon.xpi
```
Comment (pdehaan) [inline bin/build-addon.sh]: Should we add `signed-addon.xpi` to .gitignore?

Comment (wresuolc) [inline bin/build-addon.sh]: sure, that's a good idea

**Comment by wresuolc** [comment]
Waiting to hear back on who owns notes@m.c on AMO so we can take it over.

**Comment by vladikoff** [comment]
@wresuolc we can use something else if it's hard to obtain that id *shrug*

**Comment by Natim** [comment]
@wresuolc it is me :)

**Comment by wresuolc** [comment]
Thanks.  Looks good now.  I'm going to land it

**Comment by vladikoff** [comment]
@wresuolc Hero!

**Pull request merged**